src/Logger/Logger.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\password_enhancements\Logger;
+
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Logger\RfcLoggerTrait;
+use Drupal\Core\Utility\Error;
+use Exception;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Defines module specific logger.
+ */
+final class Logger implements LoggerInterface {
+
+  use RfcLoggerTrait;
+
+  /**
+   * Logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $loggerChannel;
+
+  /**
+   * Constructs a new logger service.
+   *
+   * @param string $channel
+   *   Logger channel.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_channel_factory
+   *   Logger channel factory.
+   */
+  public function __construct(string $channel, LoggerChannelFactoryInterface $logger_channel_factory) {
+    $this->loggerChannel = $logger_channel_factory->get($channel);
+  }
+
+  /**
+   * Logs exception.
+   *
+   * @param \Exception $exception
+   *   The exception that needs to be logged.
+   */
+  public function exception(Exception $exception) {
```

Comment:
1. Missing `void` return type.
2. I'd call it `logException` because it describes in a better way what this _command method_ does.
3. I would also add an optional `$message` parameter, just like `watchdog_exception()` which would allow to re-format or extend the logged exception message if needed.

Reply:
I would rather not allow changing the message, at least for now, because this is already a module specific logger, so it does not have to be too generic.

Reply:
1. and 2. fixed.

Reply:
https://github.com/Pronovix/password_enhancements/pull/1#discussion_r299315048 this comment is related to 3rd point. I hope you see why I think a customizable message (which allows this prefixing) would be beneficial. A logged exception is only useful if it helps when you have to debug something, for example from an on-prem install where you only get access to the plain text logs ;)

Reply:
The exception's error message and its stack trace is also logged, anything more than this is not really required to be able to identify the issue.

Reply:
It is logged, but if you would like to find a log entry in a listing page (like Recent log messages), good luck with finding the one that you need to debug a bug in a call when you see 10 or more similar truncated messages in the list. End even if you can do full-text search within the log entries because the same type of exceptions have the same truncated message visible and the stack trace of different exceptions could be similar you won't find what you are looking. Only if you have a specific message logged, like "Policy could not be loaded in the registration when it should" and from that the type of the logged exception matters less but the stack trace still helps you to identify what went wrong.

Reply:
I still don't see why it is necessary, but okay let it be.

Reply:
The subject should have been optional as I originally suggested. #7